### PR TITLE
Resolves #1900 - Adds ability to filter based on type of repository

### LINKED ISSUE #1900: Provide additional filter params for extras.GitRepository object
### As ...

Austin - Network Automation Engineer

### I want ...

The ability to filter the GitRepository UI and API views by additional fields

### So that ...

I can limit the choices returned in a view

### I know this is done when...

- I can filter by repository provider_contents

### Optional - Feature groups this request pertains to.

- [X] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [X] Other (not directly a platform feature)

### Database Changes

None

### External Dependencies

None

### PR DESCRIPTION
**Pull request by whitej6**
# Closes: #1900 
# What's Changed

Add `provided_contents` to filterset

# TODO

- [ ] Explanation of Change(s)
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by whitej6** [comment]
Also noticed there is not a filterform for the Table/List view for extras.GitRepository. Should I implement as well?

File: nautobot/extras/filters.py [hunk 1]
```diff
@@ -496,7 +496,7 @@ class GitRepositoryFilterSet(NautobotFilterSet):
 
     class Meta:
         model = GitRepository
-        fields = ["id", "name", "slug", "remote_url", "branch"]
+        fields = ["id", "name", "slug", "remote_url", "branch", "provided_contents"]
```
Comment (glennmatthews) [inline nautobot/extras/filters.py:499]: Ugh, we have no testing of `GitRepositoryFilterSet` in `nautobot/extras/tests/test_filters.py`.  :-( Would you be willing to take a stab at adding a few tests?

Comment (whitej6) [reply nautobot/extras/filters.py:499]: Will do

Comment (whitej6) [reply nautobot/extras/filters.py:499]: @glennmatthews done :-)

**Review by glennmatthews — APPROVED** [review]
🏆

**Pull request merged**